py_mmd_tools/mmd_to_nc.py

```diff
@@ -0,0 +1,256 @@
+
+        # Loop on expected MMD elements
+        # ie from mmd_elements.yaml = kind-off mapping between acdd and mmd
+        for mmd_element in self.mmd_yaml:
+
+            elements = self.tree.findall('mmd:' + mmd_element, self.namespaces)
+
+            if len(elements) == 0:
+                print(f'{mmd_element} not found in input MMD file')
+                continue
+
+            # Loop on  MMD elements found
+            for elem in elements:
+
+                tag = ET.QName(elem).localname
+
+                if mmd_element in ['title', 'abstract']:
+                    match = self.get_title_abstract(elem, tag)
+                    acdd = self.update_acdd(acdd, match)
+
+                elif mmd_element == 'keywords':
+                    match, sep = self.get_keyword(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                elif mmd_element == 'last_metadata_update':
+                    match, sep = self.get_last_metadata_update(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                elif mmd_element == 'metadata_identifier':
+                    match = self.get_metadata_identifier(elem)
+                    acdd = self.update_acdd(acdd, match)
+
+                elif mmd_element == 'personnel':
+                    match, sep = self.get_personnel(elem)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                # dataset_citation = repeat of mmd fields already processed elsewhere
+                elif mmd_element == 'dataset_citation':
+                    continue
+
+                # No subselements
+                elif len(list(elem)) == 0:
+                    match, sep = self.process_elem(elem, self.mmd_yaml, tag)
+                    acdd = self.update_acdd(acdd, match, sep)
+
+                # Subselements processed independently
+                else:
+
+                    for subelem in list(elem):
+
+                        subtag = ET.QName(subelem).localname
+
+                        if len(list(subelem)) > 0:
+                            for subsubelem in list(subelem):
+                                subsubtag = ET.QName(subsubelem).localname
+                                match, sep = self.process_elem(subsubelem,
+                                                               self.mmd_yaml[tag][subtag],
+                                                               subsubtag)
+                                acdd = self.update_acdd(acdd, match, sep)
+                        else:
+                            match, sep = self.process_elem(subelem, self.mmd_yaml[tag], subtag)
+                            acdd = self.update_acdd(acdd, match, sep)
+
+        # Conventions
+        acdd['Conventions'] = 'CF-1.7, ACDD-1.3'
+
+        # Open netcdf file for reading and appending
+        with nc.Dataset(self.nc, 'a') as f:
+
+            # Add all global metadata to netcdf at once
+            f.setncatts(acdd)
```

Comment:
When i do that, I erase all the netcdf global attributes that also exist in the new acdd dictionary. 
I'm not so sure that is correct. But if I don't erase, what would be the right thing to do? Merge the metadata from nc and acdd? I don't think that's right either.
Any thoughts?

Reply:
I would read the global attributes from the original and reinsert them together with the new ones. No reason to delete something unless it has the same attribute name. In that case, we may have a conflict, and we should raise an error.